### ISSUE TITLE
Markdownlint errors showing in wrong editor tabs

### ISSUE THREAD

**Issue by dprothero** [DavidAnson/vscode-markdownlint#189]
I have a hunch this started after a recent upgrade of VSCode to version 1.60.0. When a legitimate issue is detected in one editor window, for example:

![image](https://user-images.githubusercontent.com/1897006/132766634-ec14e907-fde8-431c-ba6a-8aeffbfde4a9.png)

When I switch to other markdown editor tabs, the same exact error(s) are shown on the same lines of the new tab's file contents, even though there aren't actually any issues with that file:

![image](https://user-images.githubusercontent.com/1897006/132766782-02486298-e985-4e2c-b5f3-77a0153b0507.png)

**Comment by DavidAnson** [comment]
Are you able to reproduce in an earlier version of VS Code? Or can you share the specific steps to reproduce this?

**Comment by DavidAnson** [comment]
FYI, I tried briefly on VS Code 1.60.0 on Windows and do NOT see this behavior when switching between Markdown documents.

**Comment by dprothero** [comment]
@DavidAnson agreed. I started a new project and I did NOT see the behavior either. Am working to see if I can find out what's unique about the project that it's happening in and reproduce in a simpler test case.

**Comment by dprothero** [comment]
@DavidAnson try it with markdown frontmatter

**Comment by dprothero** [comment]
@DavidAnson Here's a repo that reproduces the issue if you open it in VSCode:

https://github.com/dprothero/markdownlint-issue-189

The bizarre thing is that the issue doesn't occur if `.markdownlint-cli2.js` isn't present. I thought it was one of my settings, but even with exporting an empty object, the issue occurs.

**Comment by dprothero** [comment]
No... there's something more transitory happening here. After closing/reopening VSCode, I can't get it to occur again.

**Comment by dprothero** [comment]
@DavidAnson sorry for the churn on this. It seems my initial repro case wasn't quite right. I needed to quit VSCode in between editing the `.markdownlint-cli2.js` to narrow down the particular configuration element that was triggering the behavior. This repo now seems to reliably trigger the behavior even after quitting/restarting VSCode:

https://github.com/dprothero/markdownlint-issue-189

**Comment by dprothero** [comment]
Specifically, it seems to be this `globs` configuration that triggers the behavior:

```js
module.exports = {
  // The second "*.md" grabs markdown like README.md in the root
  globs: ["docs/**/*.{md,mdx,markdown}", "*.md"],
};
```

**Comment by DavidAnson** [comment]
Thanks for the troubleshooting! I'll try to have a look this evening.

**Comment by dprothero** [comment]
I just tried it in VSCode 1.59.1 and the issue still exists.

**Comment by DavidAnson** [comment]
There is a tiny chance this is fixed by a change I already made for the next release of the extension (coming soon).

**Comment by DavidAnson** [comment]
Ooookay. This "works" in the next release of the extension, but I wanted to be sure I understood why because my first idea was wrong. It turns out, it only gives the expected behavior in the newest extension code because it's triggering a bug in the latest version of markdownlint-cli2's support for virtual file systems. So I need to fix that and release a patch. And once I do, this will break in the newest extension code like you're reporting, but now I know why and so I'll need to fix that. Then this will work for real. Great issue, thanks for narrowing it down - the JS config file is key.
